storage/database/src/main/java/de/danoeh/antennapod/storage/database/DBWriter.java

```diff
@@ -632,6 +565,83 @@ private static void moveQueueItemHelper(final int from,
         adapter.close();
     }
 
+    public static Future<?> moveQueueItemsToTop(final Context context, final List<FeedItem> items) {
+        return runOnDbThread(() -> moveQueueItemsSynchronous(context, true, items));
+    }
+
+    public static Future<?> moveQueueItemsToBottom(final Context context, final List<FeedItem> items) {
+        return runOnDbThread(() -> moveQueueItemsSynchronous(context, false, items));
+    }
+
+    private static void moveQueueItemsSynchronous(final Context context, final boolean moveToTop,
+                                                  final List<FeedItem> items) {
+
+        if (items.isEmpty()) {
+            return;
+        }
+
+        final PodDBAdapter adapter = PodDBAdapter.getInstance();
+        adapter.open();
+        final List<FeedItem> queue = DBReader.getQueue();
+
+        if (moveToTop) {
+            Collections.reverse(items);
+        }
+
+        boolean queueModified = false;
+        List<QueueEvent> events = new ArrayList<>();
+
+        queue.removeAll(items);
+        events.add(QueueEvent.setQueue(queue));
+
+        for (FeedItem item : items) {
+            int newIndex = moveToTop ? 0 : queue.size();
+            queue.add(newIndex, item);
+            events.add(QueueEvent.moved(item, newIndex));
+            queueModified = true;
+        }
+
+        if (queueModified) {
+            adapter.setQueue(queue);
+            for (QueueEvent event : events) {
+                EventBus.getDefault().post(event);
+            }
+        } else {
+            Log.w(TAG, "moveToTop: " + moveToTop +  " - Queue was not modified.");
+        }
+        adapter.close();
+    }
+
+    public static Future<?> toggleFavoriteItem(final FeedItem item) {
```

Comment:
Please leave unrelated methods where they were. Otherwise this will make it harder to work with the git history (`git blame` to find out who added the line and why it was added would then show your commit instead of the actual commit that added the line)